AirSkyBoat/AirSkyBoat
# Fix for negative modifiers causing skill level underflow

Negative skill modifiers should no longer cause an underflow when being applied to the character's working skills table.

**_I affirm:_**


## Please enter a player-facing description

Equipment with negative skill modifiers should no longer cause that skill to go below 0.

## What does this pull request do? (Please be technical)

This is a fix for HorizonXI issue found here: https://github.com/HorizonFFXI/HorizonXI-Issues/issues/1824
Was confirmed as a bug in both ASB and LSB.

The fix will check the value that is intended to be set on the player's working skills table and make sure it's >= 0. If the value is below 0, it will set it to 0 first and then it will be assigned to the table. The changes can be found in the `BuildingCharSkillsTable` function.

## Steps to test these changes

To Test Main Job:
1) Set main job to one that has the shield skill and can equip Shiva's Shotel
2) Set your shield skill to any value <= 4
3) Give yourself the item Shiva's Shotel and equip it
4) Check your combat skills. If fixed it will be at 0, but if not it will underflow to a value around 32763

To Test Sub Job:
1) Set main job to one that DOES NOT have the shield skill and can equip Shiva's Shotel (DRK works)
2) Set your sub job to one that does have the shield skill
3) Set your shield skill to any value <= 4
4) Give yourself the item Shiva's Shotel and equip it
5) Check your combat skills. If fixed it will be at 0, but if not it will underflow to a value around 32763

To test no shield skill at all:
1) Set both your main job and sub job to one that doesn't have the shield skill (DRK/DRK does work)
2) Set your shield to any value <= 4
3) Give yourself the item Shiva's Shotel and equip it
4) Check your combat skills. If fixed it will be at 0 or not displayed, but if not it will underflow to a value around 32763

## Special Deployment Considerations

None, other than recompiling xi_map.
